# Set Hypothesis

Null hypothesis: The duration of a trip with the new bus route is same or longer than that with the original bus route. The significance level $p$= 0.05. 

Alternative hypothesis: The duration of a trip with the new bus route is significant shorter than that with the original bus route.

$H_0$: $T_1$ $>=$ $T_2$

$H_1$: $T_1$ < $T_2$

$T_1$: the duration of the new bus route

$T_2$: the duration of the original bus route


$α$ = 0.05

# Download Data

In [3]:
from __future__ import print_function

import os
import pandas as pd
import numpy as np

In [4]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > NewBusTime.CSV

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   7902      0 --:--:-- --:--:-- --:--:--  7912


In [5]:
!mv NewBusTime.CSV $PUIDATA

In [45]:
!ls $PUIDATA

201502-citibike-tripdata.csv  201507-citibike-tripdata.csv  NewBusTime.CSV
201502-citibike-tripdata.zip  201507-citibike-tripdata.zip  SATresults.CSV


# Calculate Z-test and analysis

$\mu_{pop}$=36

$\sigma_{pop}$ = 6

In [54]:
BusTime = pd.read_csv('%s/NewBusTime.CSV'%os.getenv('PUIDATA'))
value = BusTime.values

In [53]:
mu_pop = 36
sigma_pop = 6
mu_sample = value.mean()
N = len(BusTime)
sqrtN = np.sqrt(N)
Zscore = (mu_pop-mu_sample)/(sigma_pop/sqrtN)
print(Zscore)

2.49594559226


The significance level $p$ = 0.05 is corresponding to two standard devistion (1.96). The Z-test results 2.5 is farther than 2 sigmas away from the original mean, which indicates that we have 95% confidence to say that the new bus route has same or longer duration than the original bus route is not true. So reject NULL hypothesis.